File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] arr) {
        int n=arr.length;
        int left=0;
        int right=n-1;
        int maxwater=Integer.MIN_VALUE;
        while(left<right){
            int height=Math.min(arr[left],arr[right]);
            int width=right-left;
            int watertrapped=height*width;
            maxwater=Math.max(watertrapped,maxwater);
            if(arr[left]<arr[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return maxwater;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n=height.size();
        int i=0,j=n-1,maxWater=0;
        while(i!=j){
            int minHeight=min(height[i],height[j]);
            int width=j-i;
            int totalWaterTrapped=minHeight*width;
            maxWater=max(maxWater,totalWaterTrapped);
            if(height[i]<height[j]) i++;
            else j--;
        }
        return maxWater;
    }
};